### ISSUE TITLE
training loss problem

### ISSUE THREAD

**Issue by gaodihe** [google/uis-rnn#10]
During the fit process,the loss has 3 parts.Could anyone tell me what's the meaning of loss2?And why counts the number of non-zero(the code is just above calculating loss2)?

**Comment by AnzCol** [comment]
Loss2 is the negative log probability of the prior p(\sigma2). If you choose to train \sigma2 this term will matter. The reason we count non-zeros is that since rnn_truth is a collection of different length sequences padded with zeros to fit them into a tensor. We are just counting the sum of 'real lengths' of all sequences.

**Comment by wq2012** [comment]
Thanks Aonan for explaining. Closing this issue now. Please re-open it if you have more questions.
